syntax = "proto3";

package leapfrog;

// CompletionRequest is the payload to request completion
message CompletionRequest {
    string prompt = 1;
    string suffix = 2;
    int32 max_tokens = 3;
    float temperature = 4;
    float top_p = 5;
    int32 n = 6;
    bool stream = 7;
    int32 logprobs = 8;
    bool echo = 9;
    string stop = 10;  // You can only represent Union[str, list] as a string. 
    float presence_penalty = 11;
    float frequence_penalty = 12;
    int32 best_of = 13;
    map<string, float> logit_bias = 14;  // Maps are represented as a pair of a key type and a value type.
}

// CompletionRespones are what's returned by the gRPC service
message CompletionResponse {
    string completion = 1;
}

service GenerateService {
    rpc Complete (CompletionRequest) returns (CompletionResponse);
}
